lib/spack/spack/test/spec_syntax.py

```diff
@@ -631,42 +631,60 @@ def test_spec_by_hash_tokens(text, tokens):
 
 
 @pytest.mark.db
-def test_spec_by_hash(database):
+def test_spec_by_hash(database, monkeypatch, mutable_empty_config):
```

Comment:
See https://github.com/spack/spack/pull/33385#discussion_r1040036318 for more context. This will get things passing on Windows. Concretizing `a` vs `b` doesn't matter w.r.t. this failure, so if `a` was important, you can restore that after adding this suggestion.
```suggestion
def test_spec_by_hash(database, monkeypatch, mutable_empty_config, archspec_host_is_spack_test_host):
```